src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource.cs

```diff
@@ -110,6 +97,35 @@ private sealed partial class TagSource : ForegroundThreadAffinitizedObject
 
             #endregion
 
+            #region Protected by _gate
+
+            private readonly object _gate = new();
+
+            /// <summary>
+            /// Token that is triggered when we are disposed.  Used to ensure that the initial work we
+            /// kick off still gets stopped if the created tagger is disposed before that finishes.
+            /// </summary>
+            private readonly CancellationTokenSource _disposalTokenSource = new();
+
+            /// <summary>
+            /// Series of tokens used to cancel previous outstanding work when new work comes in. Also used as the lock
+            /// to ensure threadsafe writing of _eventWorkQueue.  Use <see cref="_gate"/> to access this to ensure we don't
+            /// access a disposed instance.
```

Comment:
❗ This comment says `_cancellationSeries` is the gate, and also says `_gate` is the gate.

Reply:
Stale comment. Will fix